src/translib/transformer/xfmr_ztp.go

```diff
@@ -0,0 +1,339 @@
+	}
+	return output, nil
+}
+
+/* Function to populate ztp status data structure with the status info from host service */
+
+func getZtpStatusFromHost(statusCache * ztpStatusCache, hostData map[string] interface{}) {
+
+    temp := hostData
+    for attr,val := range temp {
+	switch attr {
+	    case ZTP_STATUS_ADMIN_MODE:
+		statusCache.ztpStatusMap[ZTP_STATUS_ADMIN_MODE] = fmt.Sprintf("%t",val)
+	    case ZTP_STATUS_SERVICE:
+    		statusCache.ztpStatusMap[ZTP_STATUS_SERVICE] = fmt.Sprintf("%v",val)
+	    case ZTP_STATUS_STATUS:
+    		statusCache.ztpStatusMap[ZTP_STATUS_STATUS] = fmt.Sprintf("%v",val)
+	    case ZTP_STATUS_SOURCE:
+    		statusCache.ztpStatusMap[ZTP_STATUS_SOURCE] = fmt.Sprintf("%v",val)
+	    case ZTP_STATUS_RUNTIME:
+    		statusCache.ztpStatusMap[ZTP_STATUS_RUNTIME] = fmt.Sprintf("%v",val)
+	    case ZTP_STATUS_TIMESTAMP:
+    		statusCache.ztpStatusMap[ZTP_STATUS_TIMESTAMP] = fmt.Sprintf("%v",val)
+	    case ZTP_STATUS_JSON_VERSION:
+    		statusCache.ztpStatusMap[ZTP_STATUS_JSON_VERSION]= fmt.Sprintf("%v",val)
+	    case ZTP_STATUS_ACTIVITY_STRING:
+    		statusCache.ztpStatusMap[ZTP_STATUS_ACTIVITY_STRING] = fmt.Sprintf("%v",val)
+	    case ZTP_STATUS_ERROR:
+    		statusCache.ztpStatusMap[ZTP_STATUS_ERROR] = fmt.Sprintf("%v",val)
+	    default:
+		log.Info("Invalid attr:",attr)
+	}
+   }
+
+}
+
+/* Function to populate ztp status config section data structure with status info from host service */
+
+func getConfigSection(sectionName string, dataMap map[string]interface{}, statusCache *ztpStatusCache) {
+    statusCache.ztpConfigSectionMap[sectionName] = make(map[string]string)
+    for attr,val := range dataMap {
+	switch attr {
+	    case ZTP_CONFIG_SECTION_NAME:
+    		statusCache.ztpConfigSectionMap[sectionName][ZTP_CONFIG_SECTION_NAME] = sectionName
+	    case ZTP_CONFIG_SECTION_STATUS:
+    		statusCache.ztpConfigSectionMap[sectionName][ZTP_CONFIG_SECTION_STATUS]  = fmt.Sprintf("%v",val)
+            case ZTP_CONFIG_SECTION_RUNTIME:
+    		statusCache.ztpConfigSectionMap[sectionName][ZTP_CONFIG_SECTION_RUNTIME] = fmt.Sprintf("%v",val)
+	    case ZTP_CONFIG_SECTION_TIMESTAMP:
+    		statusCache.ztpConfigSectionMap[sectionName][ZTP_CONFIG_SECTION_TIMESTAMP] = fmt.Sprintf("%v",val)
+	    case ZTP_CONFIG_SECTION_EXITCODE:
+    		statusCache.ztpConfigSectionMap[sectionName][ZTP_CONFIG_SECTION_EXITCODE]  = fmt.Sprintf("%d",val)
+	    case ZTP_CONFIG_SECTION_IGNORE_RESULT:
+                statusCache.ztpConfigSectionMap[sectionName][ZTP_CONFIG_SECTION_IGNORE_RESULT] = fmt.Sprintf("%t",val)
+	    case ZTP_STATUS_ERROR:
+                statusCache.ztpConfigSectionMap[sectionName][ZTP_STATUS_ERROR] = fmt.Sprintf("%v",val)
+	    default:
+		log.Info("Invalid attr:",attr)
+	}
+    }
+}
+
+/* Populate ztp status ygot tree */
+
+func populateStatusYgotTree(statusObj *ocbinds.OpenconfigZtp_Ztp_State, statusCache *ztpStatusCache) {
+
+    if _,present :=statusCache.ztpStatusMap[ZTP_STATUS_ACTIVITY_STRING]; present { 
```

Comment:
`if act, present := statusCache... ; present {`

Otherwise, you are just repeating yourself.